client/src/main/java/se/fortnox/reactivewizard/client/JsonArrayDeserializer.java

```diff
@@ -0,0 +1,69 @@
+package se.fortnox.reactivewizard.client;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.async.ByteArrayFeeder;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import com.fasterxml.jackson.databind.util.TokenBuffer;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import se.fortnox.reactivewizard.util.ReflectionUtil;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JsonArrayDeserializer {
+    private final ObjectReader reader;
+    private final JsonParser parser;
+    private final ByteArrayFeeder inputFeeder;
+
+    public JsonArrayDeserializer(ObjectMapper objectMapper, Method method) {
+        Type type = ReflectionUtil.getTypeOfObservable(method);
+
+        JavaType javaType = TypeFactory.defaultInstance().constructType(type);
+        reader   = objectMapper.readerFor(javaType);
+        try {
+            parser = objectMapper.getFactory().setCodec(objectMapper).createNonBlockingByteArrayParser();
+            inputFeeder = (ByteArrayFeeder) parser.getNonBlockingInputFeeder();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public Publisher<?> process(byte[] buf) {
+        try {
+            inputFeeder.feedInput(buf, 0, buf.length);
```

Comment:
I don't see why that would be good...? It would mean that the buf would be allocated for a longer time and could not be GC'd as soon as it can now.

I am also considering putting the ByteBufFlux here instead for better memory performance, and then it must be consumed within the call, and you cannot do a defer.